mlflow/transformers.py

```diff
@@ -983,14 +1019,44 @@ def _generate_base_flavor_configuration(
 
     flavor_configuration = {
         _TASK_KEY: task,
-        _INSTANCE_TYPE_KEY: _get_instance_type(model),
-        _MODEL_PATH_OR_NAME_KEY: _get_base_model_architecture(model),
-        _PIPELINE_MODEL_TYPE_KEY: _get_instance_type(model.model),
+        _INSTANCE_TYPE_KEY: _get_instance_type(pipeline),
+        _MODEL_PATH_OR_NAME_KEY: _get_base_model_architecture(pipeline),
+        _PIPELINE_MODEL_TYPE_KEY: _get_instance_type(pipeline.model),
     }
 
+    # Extract and add to the configuration the scalar serializable arguments for pipeline args
+    for arg_key in _METADATA_PIPELINE_SCALAR_CONFIG_KEYS:
+        entry = _get_scalar_argument_from_pipeline(pipeline, arg_key)
+        if entry:
+            flavor_configuration[arg_key] = entry
+
+    # Extract a serialized representation of torch_dtype if provided
+    torch_dtype = _extract_torch_dtype_if_set(pipeline)
+    if torch_dtype:
+        flavor_configuration[_TORCH_DTYPE_KEY] = torch_dtype
+
     return flavor_configuration
 
 
+def _get_scalar_argument_from_pipeline(pipeline, arg_key):
+    """
+    Retrieve provided pipeline arguments for the purposes of instantiating a pipeline object upon
+    loading.
+    """
+
+    return getattr(pipeline, arg_key, None)
+
+
+def _extract_torch_dtype_if_set(pipeline):
+    """
+    Extract the torch datatype argument if set and return as a string encoded value.
+    """
+    torch_dtype = getattr(pipeline, _TORCH_DTYPE_KEY, None)
+
+    if torch_dtype:
+        return str(torch_dtype)
```

Comment:
```suggestion
    if torch_dtype := getattr(pipeline, _TORCH_DTYPE_KEY, None):
        return str(torch_dtype)
```

nit

Reply:
Sorry I didn't remember to use our favorite walrus ops :) Changing!

Reply:
Added in 2 other places too in _generate_base_flavor_configuration to keep the code cleaner